internal/commands/push/command.go

```diff
@@ -97,6 +97,20 @@ func (cmd *Command) Handler(profile *cli.Profile, ui terminal.UI, clients cli.Cl
 		return err
 	}
 
+	var uploadPathDependencies string
+	var dependenciesDiff realm.DependenciesDiff
+	if cmd.inputs.IncludeDependencies {
+		uploadPathDependencies, err = local.PrepareDependencies(app, ui)
+		if err != nil {
+			return err
+		}
+		dependenciesDiff, err = clients.Realm.DiffDependencies(appRemote.GroupID, appRemote.AppID, uploadPathDependencies)
+		if err != nil {
+			return err
+		}
+	}
+	defer os.Remove(uploadPathDependencies) //nolint:errcheck
```

Comment:
i feel like this `defer` should be moved after the first `err` check inside the `IncludeDependencies` block